pkg/bgpv1/gobgp/server.go

```diff
@@ -75,7 +77,22 @@ type ServerWithConfig struct {
 func NewServerWithConfig(ctx context.Context, startReq *gobgp.StartBgpRequest) (*ServerWithConfig, error) {
 	logger := NewServerLogger(log.Logger, startReq.Global.Asn)
 
-	s := server.NewBgpServer(server.LoggerOption(logger))
+	// Took from GoBGPd's default value
+	// https://github.com/osrg/gobgp/blob/44065abbca9216ee68d6747287db64b5fdc63670/cmd/gobgpd/main.go#L142
+	maxSize := 256 << 20
+	grpcOpts := []grpc.ServerOption{
+		grpc.MaxRecvMsgSize(maxSize),
+		grpc.MaxSendMsgSize(maxSize),
+	}
+
+	// Open gRPC interface for better operational state visibility. Use the UNIX domain socket
+	// to avoid access from the remote.
+	s := server.NewBgpServer(
+		server.GrpcListenAddress("unix://"+option.Config.RunDir+"/gobgp.sock"),
```

Comment:
@YutaroHayakawa 

this method can be called more then once, when multiple BGP servers are created on the host. Wont this fail in that case? Since the unix socket being bound is not unique?

Reply:
Ah, that's a good point. But, seems like we always stop the old server before creating a new server, so no problem?

https://github.com/cilium/cilium/blob/master/pkg/bgpv1/gobgp/reconcile.go#L159

Reply:
There can be multiple "ServerWithConfig"s registered: 
https://github.com/cilium/cilium/blob/5595e622243948f74187b449186e4575f451b9e5/pkg/bgpv1/gobgp/routermgr.go#L37

The `CiliumBGPPeeringPolicy` allows for you to define *1 or more* "virtual routers". 

If the `CiliumBGPPeeringPolicy` specifies two virtual routers, one with local ASN 24 and another with 25, the bind for the unix socket will fail for one of them.

Reply:
Ah, ok. Sorry, I was confused about that spec. Yep, encoding ASN sounds like a plan. Let me fix it.